sqlmesh/core/engine_adapter/base.py

```diff
@@ -2006,6 +2015,41 @@ def _truncate_table(self, table_name: TableName) -> None:
         table = exp.to_table(table_name)
         self.execute(f"TRUNCATE TABLE {table.sql(dialect=self.dialect, identify=True)}")
 
+    def _delete_then_insert_by_key(
+        self,
+        target_table: TableName,
+        source_table: QueryOrDF,
+        columns_to_types: t.Optional[t.Dict[str, exp.DataType]],
+        key: t.Sequence[exp.Expression],
+        is_unique_key: bool,
+    ) -> None:
```

Comment:
I think it should be possible to have this support Databrick's replace where right? That would allow it to still not rely on transactions.

Reply:
Added support for RELACE WHERE. Please let me know if I did this correctly.